api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TXCurrQueries.java

```diff
@@ -306,4 +322,37 @@ public static String getPatientWithoutScheduledDrugPickupDateMasterCardAmdArtPic
         returnVisitDateConcept,
         returnVisitDateForArvDrugConcept);
   }
+
+  public static String getPatientWhoAfterMostRecentDateHaveDrusPickupOrConsultation(
+      int adultoSeguimentoEncounterType,
+      int aRVPediatriaSeguimentoEncounterType,
+      int aRVPharmaciaEncounterType,
+      int masterCardDrugPickupEncounterType,
+      int artDatePickup) {
+    String query =
+        " select p.patient_id "
+            + " from patient p "
+            + " inner join encounter e on  e.patient_id = p.patient_id "
+            + " where  p.voided=0  and e.voided=0 "
+            + " and e.encounter_type in (%s,%s,%s)  "
+            + " and e.encounter_datetime > (select  max(encounter_datetime) from encounter where  patient_id= p.patient_id ) "
+            + " and e.location_id= :location group by p.patient_id "
+            + " union "
+            + " select p.patient_id "
+            + " from patient p "
+            + " inner join encounter e on e.patient_id=p.patient_id "
+            + " inner join obs o on o.encounter_id=e.encounter_id "
+            + " where e.encounter_type = %s and p.voided=0  and e.voided=0 and o.voided=0 "
+            + " and o.concept_id=%s  "
+            + " and o.value_datetime > (select  max(value_datetime) from obs where  encounter_id= e.encounter_id) "
```

Comment:
@cfaife this inner query is not filtering on obs.concept_id - is that intentional?